lib/concurrent/synchronized_object.rb

```diff
@@ -0,0 +1,243 @@
+module Concurrent
+
+  # Safe synchronization under any Ruby implementation
+  # Provides a single layer which can improve its implementation over time without changes needed to
+  # the classes using it. Use {SynchronizedObject} not this abstract class.
+  # @example
+  #   class AnClass < SynchronizedObject
+  #     def initialize
+  #       super
+  #       synchronize { @value = 'asd' }
+  #     end
+  #
+  #     def value
+  #       synchronize { @value }
+  #     end
+  #   end
+  class AbstractSynchronizedObject
```

Comment:
I really like this abstraction a lot. :+1: I love that this is the result of the gem's evolution--a pattern emerged over time and we refactored into this. Thank you very much for creating this!